rust/operator-binary/src/crd/mod.rs

```diff
@@ -103,85 +104,93 @@ pub enum Error {
-
-    /// Kafka settings that affect all roles and role groups.
-    /// The settings in the `clusterConfig` are cluster wide settings that do not need to be configurable at role or role group level.
-    pub cluster_config: KafkaClusterConfig,
+#[versioned(version(name = "v1alpha1"))]
+pub mod versioned {
+    /// A Kafka cluster stacklet. This resource is managed by the Stackable operator for Apache Kafka.
+    /// Find more information on how to use it and the resources that the operator generates in the
+    /// [operator documentation](DOCS_BASE_URL_PLACEHOLDER/kafka/).
+    #[versioned(k8s(
+        group = "kafka.stackable.tech",
+        kind = "KafkaCluster",
+        plural = "kafkaclusters",
+        status = "KafkaClusterStatus",
+        shortname = "kafka",
+        namespaced,
+        crates(
+            kube_core = "stackable_operator::kube::core",
+            k8s_openapi = "stackable_operator::k8s_openapi",
+            schemars = "stackable_operator::schemars"
+        )
+    ))]
+    #[derive(Clone, CustomResource, Debug, Deserialize, JsonSchema, Serialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct KafkaClusterSpec {
+        // no doc - docs in ProductImage struct.
+        pub image: ProductImage,
+
+        // no doc - docs in Role struct.
+        pub brokers: Option<Role<KafkaConfigFragment, GenericRoleConfig, JavaCommonConfig>>,
+
+        /// Kafka settings that affect all roles and role groups.
+        /// The settings in the `clusterConfig` are cluster wide settings that do not need to be configurable at role or role group level.
+        pub cluster_config: v1alpha1::KafkaClusterConfig,
+
+        // no doc - docs in ClusterOperation struct.
+        #[serde(default)]
+        pub cluster_operation: ClusterOperation,
+    }
 
-    // no doc - docs in ClusterOperation struct.
-    #[serde(default)]
-    pub cluster_operation: ClusterOperation,
+    #[derive(Clone, Deserialize, Debug, Eq, JsonSchema, PartialEq, Serialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct KafkaClusterConfig {
+        /// Authentication class settings for Kafka like mTLS authentication.
+        #[serde(default)]
+        pub authentication: Vec<KafkaAuthentication>,
+
+        /// Authorization settings for Kafka like OPA.
+        #[serde(default)]
+        pub authorization: KafkaAuthorization,
+
+        /// TLS encryption settings for Kafka (server, internal).
+        #[serde(
+            default = "tls::default_kafka_tls",
+            skip_serializing_if = "Option::is_none"
+        )]
+        pub tls: Option<KafkaTls>,
+
+        /// Name of the Vector aggregator [discovery ConfigMap](DOCS_BASE_URL_PLACEHOLDER/concepts/service_discovery).
+        /// It must contain the key `ADDRESS` with the address of the Vector aggregator.
+        /// Follow the [logging tutorial](DOCS_BASE_URL_PLACEHOLDER/tutorials/logging-vector-aggregator)
+        /// to learn how to configure log aggregation with Vector.
```

Comment:
We should really fix up these doc comments to move the URLs to different lines.

Also, now that we generate cargo docs, the URLs will be invalid. So maybe we should default them, and whatever downstream thing that does a search/replace can operator on the actual basename.